app/helpers.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+use Gladiator\Models\User;
+use Gladiator\Services\Northstar\Exceptions\NorthstarUserNotFoundException;
+
+if (! function_exists('matchEmailDomain')) {
+    /**
+     * Match supplied email against a specific domain or default domain.
+     *
+     * @param  string  $email
+     * @param  string  $domain
+     * @return boolean
+     */
+    function matchEmailDomain($email, $domain = 'dosomething.org')
+    {
+        $pieces = explode('@', $email);
+
+        if (is_array($pieces) && count($pieces) > 1) {
+            return $pieces[1] === $domain ? true : false;
+        }
+
+        return false;
+    }
+}
+
+if (! function_exists('findUserInSystem')) {
+    /**
+     * @param  string  $id
+     * @param  string  $email
+     * @return \Gladiator\Models\User|string
+     * @throws \Gladiator\Services\Northstar\Exceptions\NorthstarUserNotFoundException
+     */
+    function findUserInSystem($id, $type = 'email')
```

Comment:
I was thinking about that as well, wasn't sure what was best and also for maintaining testability...